src/atopile/components.py

```diff
@@ -81,66 +55,111 @@ class NoMatchingComponent(errors.AtoError):
+    }
 
-    specd_type = _generic_to_type_map[specd_mpn]
-    filters.append(f"type == '{specd_type}'")
+    payload = {
+        **specd_data_json,
+    }
 
-    # Apply filters we know how to process
-    try:
-        value_range = specd_data["value"]
-    except KeyError as ex:
-        raise KeyError("Generics are missing data - internal error") from ex
+    return _make_api_request(name, component_addr, payload, log)
 
-    if not isinstance(value_range, Physical):
-        raise ValueError(f"Value must be a Physical, not {type(value_range)}")
 
-    # Ensure the component's value is completely contained within the specd value
+def _make_api_request(name, component_addr, payload, log):
+    url = "https://get-component-atsuhzfd5a-uc.a.run.app"
     try:
-        generic_unit = _generic_to_unit_map[specd_mpn]
-        min_float_val = (value_range.min_val * value_range.unit).to(generic_unit).magnitude
-        max_float_val = (value_range.max_val * value_range.unit).to(generic_unit).magnitude
-    except pint.DimensionalityError as ex:
-        raise errors.AtoTypeError(
-            f"{value_range.unit} cannot be converted to {generic_unit} for $addr", addr = component_addr,
-            title="Invalid unit",
-        ) from ex
-
-    filters.append(f"min_value >= {min_float_val}")
-    filters.append(f"max_value <= {max_float_val}")
-
-    # Ensure the component's footprint is correct
-    filters.append(f"Package == '{_generics_to_db_fp_map[specd_data['footprint']]}'")
-
-    # Combine filters using reduce
-    combined_filter = " & ".join(filters)
-    filtered_df = df.query(combined_filter)
-    if filtered_df.empty:
-        msg = "No component matching spec for $addr \n"
-        msg += "\n & ".join(filters)
-        raise NoMatchingComponent(msg, addr=component_addr)
+        log.debug(payload)
+        response = requests.post(url, json=payload)
+        response.raise_for_status()
+        best_component = response.json().get("Best Component")
 
-    # FIXME: Currently our cost function is dumb - it only knows dollars
-    # In the future this cost function should incorporate other things the user is
-    # likely to care about
-    idx_min = filtered_df["Price (USD)"].idxmin(skipna=True)
+        if not best_component:
+            raise NoMatchingComponent("No valid component found", addr=component_addr)
 
-    # In this case we seem to hit NaN, which implies we don't have
-    # cost info - which is really a bug in the db, but for the users' sake
-    # we'll just return the first component
-    if pd.isna(idx_min):
-        return filtered_df.iloc[0].to_dict()
+        lcsc = best_component["lcsc_id"]
+        log.debug(f"Successfully fetched component {lcsc} for {name}")
+        return best_component
 
-    return filtered_df.loc[idx_min].to_dict()
+    except requests.RequestException as e:
+        log.warning(f"API request failed: {e}")
+        return {"LCSC Part #": "Part not found", "value": "N/A", "unit": "N/A"}
```

Comment:
I'd generally recommend using an `attrs` object or [`dataclass`](https://docs.python.org/3/library/dataclasses.html) for things like this. It's really easy to get dict keys wrong and type checkers can't help you with it.

Reply:
Updated